### DISCUSSION TITLE
How do you develop connectors in the repo workspace?

### DISCUSSION THREAD

**Discussion by tisonkun** [apache/pulsar#16886] [Q&A]
That said, working with Pulsar connectors is based on Pulsar functions, and in `conf/functions_worker.yml` we configure:

```yaml
connectorsDirectory: ./connectors
```

However, the bundled connectors are under `./distribution/io/target/apache-pulsar-io-connectors-<version>-bin`.

It causes when I run code snippets from our docs like:

```shell
./bin/pulsar-admin sources create --name generator --destinationTopicName generator_test --source-type data-generator
```

It complains:

```
Invalid source type 'data-generator' -- Available sources are: []
```

Creating a directory under the root path and copying connectors' NAR files works, but it's a bit clumsy:

```shell
cp distribution/io/target/apache-pulsar-io-connectors-2.11.0-SNAPSHOT-bin/*.nar connectors 
```

I want to learn how you develop connectors in the repo workspace. Is there some way better than that described above?

cc @merlimat @lhotari

**Comment by asafm** [comment]
@nlu90 might know?

**Reply by nlu90** [reply]
Some thoughts:
1. update the `functions_worker.yml`'s `connectorsDirectory` to point to your dev directory
2. create a soft link to your directory
